server-node-new/app.js

```diff
@@ -0,0 +1,197 @@
+const createError = require("http-errors");
+const express = require("express");
+const { join } = require("path");
+const cookieParser = require("cookie-parser");
+const logger = require("morgan");
+
+const indexRouter = require("./routes/index");
+const pingRouter = require("./routes/ping");
+
+const { json, urlencoded } = express;
+
+var app = express();
+
+// Database Setup
+
+const mongoose = require("mongoose");
+const mongoDB = "mongodb://localhost/test";
+mongoose.connect(mongoDB, { useNewUrlParser: true, useUnifiedTopology: true });
+const db = mongoose.connection;
+db.once("open", function(){
+  console.log("Database connected successfully");
+}).on("error", function(err){
+  console.log("Error", err);
+});
+
+// Models
+
+const userSchema = new mongoose.Schema({
+  userName:         { type: String, index: true, unique: true, sparse: true }, // not sure if this is the standard implementation to allow null to not be unique
+  firstName:        String,
+  lastName:         String,
+  email:            { type: String, required: true }, // remove unique: true for testing
+  timezone:         String,
+  profilePicture:   String,
+  accessToken:      { type: String, required: true },
+  stripeCustomerId: String,
+  createdAt:        { type: Date, default: Date.now }, 
+  updatedAt:        { type: Date, default: Date.now },
+});
+
+const User = mongoose.model("User", userSchema);
+
+const availabilitySchema = new mongoose.Schema({
+  user:       { type: mongoose.Schema.Types.ObjectId, ref: "User" },
+  day:        { type: Number, required: true },
+  startTime:  { type: Number, required: true, default: 540 },
+  endTime:    { type: Number, required: true, default: 1020 },
+  createdAt:  { type: Date, default: Date.now }, 
+  updatedAt:  { type: Date, default: Date.now },  
+});
+
+const Availability = mongoose.model("Availability", availabilitySchema);
+
+const meetingSchema = new mongoose.Schema({
```

Comment:
Everything is currently in one big file at the moment, we probably want to split this off into its own file `models.js` and `controller` folder as well for the commented out code